mesonbuild/minstall.py

```diff
@@ -276,6 +285,27 @@ def __init__(self, options: 'ArgumentType', lf: T.TextIO):
         self.lf = lf
         self.preserved_file_count = 0
 
+        dry_run = options.dry_run
+        noop = lambda *args, **kwargs: None
+        Popen_noop = lambda *args, **kwargs: (types.SimpleNamespace(returncode=0), '', '')
+        run_exe_noop = lambda *args, **kwargs: 0
```

Comment:
I'm going to expect mypy to choke on these, and require you to not use lambdas, but I might surprised. Especially the `SimpleNamespace` thing